src/ErrorHandling/RecentErrorsPage.tsx

```diff
@@ -0,0 +1,138 @@
+import React from "react";
+import { makeStyles } from "tss-react/mui";
+import { ErrorRecord, ErrorState, killAllScripts } from "./ErrorState";
+import { useRerender } from "../ui/React/hooks";
+import { Typography, Button } from "@mui/material";
+import { Theme } from "@mui/material/styles";
+import { OptionSwitch } from "../ui/React/OptionSwitch";
+
+export function RecentErrorsPage(): React.ReactElement {
+  const rerender = useRerender();
+  React.useEffect(() => {
+    const listener = () => {
+      rerender();
+    };
+    ErrorState.ErrorUpdate.subscribe(listener);
+    ErrorState.ErrorPageOpen = true;
+    ErrorState.UnreadErrors = 0;
+  }, [rerender]);
+  const { classes } = useStyles();
+
+  const showError = (error: ErrorRecord): void => {
+    ErrorState.ErrorUpdate.emit({ ...error, force: true });
+  };
+
+  const nthIndexOf = (string: string, pattern: string, n: number) => {
+    let i = -1;
+
+    while (n-- && i++ < string.length) {
+      i = string.indexOf(pattern, i);
+      if (i < 0) break;
+    }
+    return i;
+  };
+
+  const formatMessage = (message: string): string => {
+    // Add zero-width space after each slash to allow clean wrapping
+    const cleanedMessage = message.replaceAll("/", "/​");
+    const fifthLineBreak = nthIndexOf(message, "\n", 5);
+    if (fifthLineBreak !== -1) {
+      // If the message has more than 4 line breaks, truncate it to the first 5 lines
+      return cleanedMessage.slice(0, fifthLineBreak + 1) + " ...";
+    }
+    return cleanedMessage;
+  };
```

Comment:
```suggestion
  const formatMessage = (message: string): string => {
    /**
     * - Add a zero-width space after each slash to allow clean wrapping.
     * - Replace 2+ newline characters with only 1 newline character to reduce the number of empty lines.
     */
    return message.replaceAll("/", "/\u200B").replaceAll(/\n{2,}/g, "\n");
  };
```
We only need to do this. The remaining part will be done in CSS. I'll submit a full patch later.

Reply:
suggestion and patch merged.